sklearn/inspection/_plot/decision_boundary.py

```diff
@@ -254,18 +292,32 @@ def from_estimator(
             :term:`decision_function`, :term:`predict` as the target response.
             If set to 'auto', the response method is tried in the following order:
             :term:`decision_function`, :term:`predict_proba`, :term:`predict`.
-            For multiclass problems, :term:`predict` is selected when
-            `response_method="auto"`.
+
+            .. versionchanged:: 1.6
+                For multiclass problems, 'auto' no longer defaults to 'predict'.
 
         class_of_interest : int, float, bool or str, default=None
-            The class considered when plotting the decision. If None,
-            `estimator.classes_[1]` is considered as the positive class
-            for binary classifiers. Must have an explicit value for
-            multiclass classifiers when `response_method` is 'predict_proba'
-            or 'decision_function'.
+            The class to be plotted when `response_method` is 'predict_proba'
+            or 'decision_function'. If None, `estimator.classes_[1]` is considered
+            the positive class for binary classifiers. For multiclass
+            classifiers, if None, all classes will be represented in the
+            decision boundary plot; the class with the highest response value
+            at each point is plotted. The color of each class can be set via
+            `multiclass_cmap`.
 
             .. versionadded:: 1.4
 
+        multiclass_cmap : list[str, Colormap] default=None
```

Comment:
Now that I'm looking at this, I'm thinking that the best API would be a list of strings containing primary colors:

```python
DecisionBoundaryDisplay.from_estimator(model, X, colors=["tab:blue", "tab:orange", "tab:green"])
```

I would probably rename the parameter to `colors` or `class_colors` if there is an ambiguity.
We can get the RGB representation of those standard Matplotlib color with

```python
import matplotlib as mpl
r, g, b = mpl.colors.to_rgb("tab:blue")
```

Then, the idea will be to always a `LinearSegmentedColormap` from each color.

Reply:
Basically, I think that we cover the major use cases and we can point to this tutorial: https://matplotlib.org/stable/gallery/color/named_colors.html to get the name of the colors.

Reply:
So as a short example, it would like:

```python
from sklearn.datasets import load_iris
from sklearn.linear_model import LogisticRegression
from sklearn.inspection import DecisionBoundaryDisplay

X, y = load_iris(return_X_y=True)
X = X[:, :2]

model = LogisticRegression().fit(X, y)
disp = DecisionBoundaryDisplay.from_estimator(
    model, X, multiclass_cmap=["r", "g", "b"]
)

color_map = {0: 'r', 1: 'g', 2: 'b'}
colors = [color_map[i] for i in y]
disp.ax_.scatter(X[:, 0], X[:, 1], c=colors)
```

![image](https://github.com/user-attachments/assets/f8fd2c44-5707-4edd-befb-e39bf136e7fc)

Reply:
I think this idea would be better but let me clarify what you mean.

> I would probably rename the parameter to colors or class_colors if there is an ambiguity.

Would this parameter only be used when plotting all classes in multi-class case? If that is the case, why not use the term 'multiclass' in the parameter?

If that is not the case, I think this could be confusing. Both `contour` and `contourf` have `colors` AND `cmap` parameters (only one should be set). If we want to use this term outside of the multiclass case, it would be difficult to determine whether to pass the arg to `colors` or `cmap` ?

---

I do think this idea is better because when plotting several classes, you would generally want to use a single color colormap per class, otherwise it would be visually confusing. It also may be easier as color names may be more well known than cmap names.

The cons:

* May be a little confusing as we are asking for 'colors' but we will actually be passing these on to `cmap`
* User will not be able to use their own colormap or a registered mpl cmap.

Reply:
> If that is the case, why not use the term 'multiclass' in the parameter?

I was more against using the term `cmap` indeed. I would be fine with `multiclass` appearing.

Reply:
> User will not be able to use their own colormap or a registered mpl cmap.

Thinking a bit more, we could accept registered colormap from matplotlib and do the same process that we do for `viridis`.

So the option would be a list of colors that should match `n_classes` or a colormap with a single string that we will resample to get `n_classes` colors.

Reply:
So we will need a name to reflect both. I assume that we might want to get away from `color` or `cmap` and a name that reflect better something about our display.

Reply:
Okay I've amended to list of colors or cmap str. Let me know what you think.

I named the parameter 'multiclass_colors', couldn't think of a better name and I feel that we sort of need the word 'color' because that's what we are doing... but I am open to change though!

(I will add tests etc once we are happy with API)

Reply:
Let's start with this API. We will see how it goes with the second reviewer. In the worst case scenario, it will come back to a search and replace.

Reply:
I'm resolving this discussion then.